hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitter.java

```diff
@@ -139,6 +151,13 @@ private ReducingConsumer processCommittedOffsets() {
         return committedOffsetsReducer;
     }
 
+    private void updateMaxCommittedOffsets(Map<SubscriptionPartition, Long> maxDrainedCommittedOffsets) {
+        maxDrainedCommittedOffsets.forEach((partition, drainedOffset) ->
+                maxCommittedOffsets.compute(partition, (p, storedOffset) ->
+                        storedOffset == null || storedOffset < drainedOffset ? drainedOffset : storedOffset)
+        );
```

Comment:
I'm not 100% sure what is going on here, can you clarify a bit? 😄

Reply:
OK, so there's a `map(partition->offset)`. We want this map to contain the largest offset per partition. If we have a new offset (`drainedOffset`) to be added, we need to check if the partition contained a larger offset already (`storedOffset == null || storedOffset < drainedOffset` -> it means it didn't, so we can override it). We're inserting/updating items using `compute` method here.